lib/string.js

```diff
@@ -196,6 +197,26 @@ internals.String.prototype.email = function (isEmailOptions) {
 };
 
 
+internals.String.prototype.uri = function (uriOptions) {
+    var regex = internals.uriRegex;
+
+    if (uriOptions) {
+        Hoek.assert(typeof uriOptions === 'object', 'uri options must be an object');
+        Hoek.assert(/[a-zA-Z][a-zA-Z0-9+-\.]*/.test(uriOptions.scheme), 'scheme must be a valid scheme');
+        regex = createUriRegex(uriOptions.scheme);
+    }
+
+    return this._test('uri', undefined, function (value, state, options) {
+
+        if (regex.test(value)) {
+            return null;
+        }
+
+        return Errors.create('string.uri', null, state, options);
```

Comment:
I think it would be worth having 2 messages depending on whether we have schemes so that the error is more explicit on the expectations